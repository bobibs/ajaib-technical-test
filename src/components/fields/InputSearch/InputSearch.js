import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import ImageBox from '../../elements/ImageBox';
import styles from './styles.module.css';

export default function InputSearch(props) {
  const { className, onClick, placeholder, resetValue } = props;
  const [value, setValue] = useState('');

  const classes = [styles.root, className].filter(Boolean).join(' ');
  const clearValue = () => setValue('');

  useEffect(() => {
    if (resetValue) {
      setValue('');
    }
  }, [resetValue]);

  return (
    <div className={classes}>
      <input
        className={styles.input}
        onChange={(e) => setValue(e.target.value)}
        value={value}
        placeholder={placeholder}
        type='text'
      />
      {value.length ? (
        <div className={styles.clearIcon} onClick={clearValue}>
          <ImageBox className={styles.icon} src='clear' />
        </div>
      ) : (
        <div style={{ width: '2rem' }} />
      )}
      <div className={styles.icons} onClick={() => onClick(value)}>
        <ImageBox className={styles.icon} src='search' />
      </div>
    </div>
  );
}

InputSearch.defaultProps = {
  className: '',
  onClick: () => {},
  placeholder: 'Search',
  resetValue: false,
};

InputSearch.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
  placeholder: PropTypes.string,
  resetValue: PropTypes.bool,
};
