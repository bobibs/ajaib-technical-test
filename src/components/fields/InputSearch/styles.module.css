.root {
    display: flex;
    border: 0.05rem solid #888888;
    border-radius: 0.25rem;
    height: 2rem;
}

.input {
    border: none;
    padding: 0 0.5rem;
    border-radius: 0.25rem 0 0 0.25rem;
}

.clearIcon {
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    opacity: 0.75;
    width: 2rem;
}

.icons {
    background-color: var(--primary-color);
    border-radius: 0 0.25rem 0.25rem 0;
    width: 2rem;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
}

.icon {
    height: 1rem;
    width: 1rem;
}