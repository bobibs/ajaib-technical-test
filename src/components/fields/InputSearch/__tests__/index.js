import React, { useEffect, useState } from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import InputSearch from '../InputSearch';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
  useEffect: jest.fn(),
}));

const setValue = jest.fn();

describe('src/components/elements/InputSearch', () => {
  beforeEach(() => {
    useState.mockImplementation((v) => ['value', setValue]);
    useEffect.mockImplementation((fn) => fn());
  });

  test('render', () => {
    InputSearch.defaultProps.onClick();
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<InputSearch resetValue={true} />);
    expect(tree).toMatchSnapshot();
  });

  test('function', () => {
    const event = {
      target: {
        value: 'val',
      },
    };
    const onClick = jest.fn();
    const result = InputSearch({ onClick });

    result.props.children[0].props.onChange(event);
    expect(setValue).toHaveBeenCalledWith('val');

    result.props.children[1].props.onClick();
    expect(setValue).toHaveBeenCalledWith('');

    result.props.children[2].props.onClick();
    expect(onClick).toHaveBeenCalledWith('value');
  });
});
