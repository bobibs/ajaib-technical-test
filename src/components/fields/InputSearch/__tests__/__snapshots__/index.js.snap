// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/InputSearch render 1`] = `
<div
  className="root"
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value="value"
  />
  <div
    className="clearIcon"
    onClick={[Function]}
  >
    <ImageBox
      className="icon"
      format="svg"
      onClick={[Function]}
      src="clear"
    />
  </div>
  <div
    className="icons"
    onClick={[Function]}
  >
    <ImageBox
      className="icon"
      format="svg"
      onClick={[Function]}
      src="search"
    />
  </div>
</div>
`;
